File: API/Services/CostCenterService.cs
using BudgetManagement.Api.Models;
using BudgetManagement.Api.Repositories;

namespace BudgetManagement.Api.Services;

public class CostCenterService : ICostCenterService
{
    private readonly ICostCenterRepository _costCenterRepository;
    private readonly ILogger<CostCenterService> _logger;

    public CostCenterService(ICostCenterRepository costCenterRepository, ILogger<CostCenterService> logger)
    {
        _costCenterRepository = costCenterRepository;
        _logger = logger;
    }

    public async Task<IEnumerable<CostCenter>> GetAllCostCentersAsync()
    {
        _logger.LogInformation("Retrieving all cost centers");
        return await _costCenterRepository.GetAllAsync();
    }

    public async Task<CostCenter?> GetCostCenterByIdAsync(Guid costCenterId)
    {
        _logger.LogInformation("Retrieving cost center with ID: {CostCenterId}", costCenterId);
        return await _costCenterRepository.GetByIdAsync(costCenterId);
    }

    public async Task<IEnumerable<CostCenter>> GetActiveCostCentersAsync()
    {
        _logger.LogInformation("Retrieving active cost centers");
        return await _costCenterRepository.GetActiveCostCentersAsync();
    }
}

File: UI/src/services/cost-center-service.ts
import { apiClient } from './api-client';
import { CostCenter } from '../types/budget';

/**
 * Service for cost center-related API calls
 */
export const costCenterService = {
  /**
   * Get all active cost centers
   */
  getAllCostCenters: async (): Promise<CostCenter[]> => {
    return apiClient.get<CostCenter[]>('/costcenters');
  },

  /**
   * Get cost center by ID
   */
  getCostCenterById: async (id: string): Promise<CostCenter> => {
    return apiClient.get<CostCenter>(`/costcenters/${id}`);
  },
};
